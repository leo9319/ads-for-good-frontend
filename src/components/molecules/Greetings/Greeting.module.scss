@import '../../../styles/breakpoints';

.greetings {
  display: grid;
  gap: 4px;

  &.cardContainer {
    @media (min-width: $sm-min) {
      min-width: 360px;
    }
  }

  .heading {
    font-weight: 600;
    font-size: var(--typography-font-size-8);
    line-height: var(--typography-line-height-8);
    font-family: var(--typography-default-font-family);

    @media (max-width: 767px) {
      font-size: var(--typography-font-size-7);
      line-height: var(--typography-line-height-7);
    }
  }

  .message {
    text-align: center;
    padding-block: var(--spacing-xs);
  }
}
