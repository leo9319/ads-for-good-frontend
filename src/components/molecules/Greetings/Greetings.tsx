import React, { ReactElement } from 'react';
import Text from '@radix-styles/atoms/Text';
import { Heading } from '@radix-ui/themes';
import styles from './Greeting.module.scss';
import { classNames } from '@utils/common/classNames';

/**
 * @interface
 * The Greetings props type of {@link Greetings}.
 */
export interface GreetingsProps {
  /**
   * Title of the card to show in the page
   * @property
   */
  title: string;
  /**
   * Contents to show below of the title
   * @property
   */
  content?: string;
  /**
   * Pass it to change the horizontal alignment of content
   * @property
   */
  align?: 'center' | 'left' | 'right';
  /**
   * Pass it to change the minimum width as 360px of content
   * @property
   */
  isCardContainer?: boolean;
}

/**
 * Greetings component
 *
 * @group Molecules
 * @category Component
 *
 * @param {GreetingsProps} props - The Greetings props
 * @returns {ReactElement} - Greetings Component
 */
export const Greetings = ({
  title,
  content,
  align = 'left',
  isCardContainer = false,
}: GreetingsProps): ReactElement => {
  const classes = classNames(
    styles.greetings,
    isCardContainer ? styles.cardContainer : ''
  );
  return (
    <div className={classes}>
      <Heading as="h1" align={align} className={styles.heading}>
        {title}
      </Heading>
      {content ? (
        <Text as="p" className={styles.message}>
          {content}
        </Text>
      ) : (
        ''
      )}
    </div>
  );
};

export default Greetings;
