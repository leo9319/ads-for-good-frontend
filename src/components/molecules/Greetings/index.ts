export type { GreetingsProps } from './Greetings';
export { default } from './Greetings';
export { Greetings } from './Greetings';
