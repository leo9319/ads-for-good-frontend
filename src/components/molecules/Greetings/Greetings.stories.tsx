import React from 'react';
import { Theme } from '@radix-ui/themes';
import Greetings from './Greetings';
import { StoryObj, Meta } from '@storybook/react';

const meta = {
  title: 'Library/Molecules/Greetings',
  component: Greetings,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    content: {
      control: { type: 'text' },
    },
    title: {
      control: { type: 'text' },
    },
    align: {
      control: { type: 'select' },
      options: ['left', 'center', 'right'],
    },
  },
} satisfies Meta<typeof Greetings>;

export default meta;

type Story = StoryObj<typeof meta>;

const content =
  'World Vision Canada is a Christian relief, development, and advocacy organization working to create lasting change in the lives of children and families. (To accommodate up to five lines of text).';
export const GreetingsWithSignIn: Story = {
  args: {
    content: content,
    title: 'Welcome back!',
    align: 'center',
  },
};
