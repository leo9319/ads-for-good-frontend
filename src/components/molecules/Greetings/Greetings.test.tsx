import React from 'react';
import { render } from '@testing-library/react';
import Greetings from './Greetings';

describe('Greetings', () => {
  it('Should render successfully', () => {
    const { baseElement } = render(<Greetings title="Welcome back!" />);
    expect(baseElement).toMatchSnapshot();
  });
});
