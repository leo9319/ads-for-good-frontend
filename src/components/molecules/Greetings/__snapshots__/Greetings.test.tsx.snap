// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings Should render successfully 1`] = `
<body>
  <div>
    <div
      class="greetings"
    >
      <h1
        class="rt-Heading rt-r-size-6 rt-r-ta-left heading"
      >
        Welcome back!
      </h1>
      <p
        class="rt-Text typography-font-size-3 typography-font-weight-regular typography-line-height-3 typography-font-family-default message"
      />
    </div>
  </div>
</body>
`;
